### ISSUE TITLE
Spanmetrics: request rate metrics are jagged, 0 value every minute

### ISSUE THREAD

**Issue by domasx2** [grafana/tempo#2516]
**Describe the bug**
rate over spanmetrics are suspiciously jagged, dropping to 0 every minute

![image](https://github.com/grafana/tempo/assets/847684/6d019c6c-f7a6-4b8a-a6aa-8fcb14b28aa8)

[explore in grafana ops](https://ops.grafana-ops.net/explore?orgId=1&left=%7B%22datasource%22:%22000000134%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22sum%28rate%28traces_spanmetrics_latency_count%7B%7D%20%5B$__rate_interval%5D%29%29%20by%20%28source%29%22,%22range%22:true,%22instant%22:false,%22datasource%22:%7B%22type%22:%22prometheus%22,%22uid%22:%22000000134%22%7D,%22editorMode%22:%22code%22,%22exemplar%22:false,%22format%22:%22time_series%22%7D%5D,%22range%22:%7B%22from%22:%22now-1h%22,%22to%22:%22now%22%7D%7D)

**To Reproduce**
1. Enable spanmetrics
2. Send good volume of traces
3. Do a rate and note that every minute spanmetrics timeseries produce 0 or near 0 value

Goutham:
So basically the datapoints are sent every 15s, but they are updated every 1m.
The problem is that while samples are being sent every 15s, the update of the metric happens only every minute.

For the "instant" query: `traces_spanmetrics_latency_count{__metrics_gen_instance="metrics-generator-5684fd747f-7zgwt",cluster=.........}[10m]`, you get:
<img width="611" alt="Screenshot 2023-05-30 at 17 17 59" src="https://github.com/grafana/tempo/assets/7354143/ce550cea-ed66-4741-9805-4ecf9094f424">

Here you can see that the metrics are updated only 1m, but sent every 15s.

**Comment by domasx2** [comment]
@cyrille-leclerc  I think you are correct. Ops prometheus datasource has scrape interval set to `15s`, while "standard" cloud instance have `60s`. That should explain it

**Comment by mapno** [comment]
After investigation, we got to the conclusion that the principal cause of the sawtooth like metrics when applying `rate()` is the batching and buffering since a span is created until it reaches the metrics-generator. This causes that spans will arrive in small bursts to the processors, more that they're naturally being generated.
